claims.go

```diff
@@ -140,174 +122,63 @@ func securityLifeCycleToString(v uint16) string {
-func (p *Evidence) Sign(signer *cose.Signer) ([]byte, error) {
-	if signer == nil {
-		return nil, errors.New("nil signer")
-	}
-
-	p.message = cose.NewSign1Message()
-
-	var err error
-	p.message.Payload, err = p.Claims.ToCBOR()
-	if err != nil {
-		return nil, err
-	}
-
-	alg := signer.GetAlg()
-	if alg == nil {
-		return nil, errors.New("signer has no algorithm")
-	}
-
-	p.message.Headers.Protected[1] = alg.Value
-
-	err = p.message.Sign(rand.Reader, []byte(""), *signer)
-	if err != nil {
-		return nil, err
-	}
 
-	wrap, err := cose.Marshal(p.message)
-	if err != nil {
-		return nil, err
-	}
-
-	return wrap, nil
 }
 
-// Verify verifies any attached signature for the Evidence.
-func (p *Evidence) Verify(pk crypto.PublicKey) error {
-	if p.message == nil {
-		return errors.New("token does not appear to be signed")
-	}
-
-	alg, err := cose.GetAlg(p.message.Headers)
-	if err != nil {
-		return err
-	}
-
-	verifier := cose.Verifier{
-		Alg:       alg,
-		PublicKey: pk,
-	}
-
-	err = p.message.Verify([]byte(""), verifier)
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-func (p *Claims) validate() error {
-	err := p.validateProfile()
-	if err != nil {
-		return err
-	}
-
-	err = p.validatePartitionID()
-	if err != nil {
-		return err
-	}
-
-	err = p.validateSecurityLifeCycle()
-	if err != nil {
-		return err
-	}
-
-	err = p.validateImplID()
-	if err != nil {
-		return err
-	}
-
-	err = p.validateBootSeed()
-	if err != nil {
-		return err
-	}
-
-	err = p.validateHwVersion()
-	if err != nil {
-		return err
-	}
-
-	err = p.validateSwComponents()
-	if err != nil {
-		return err
-	}
-
-	err = p.validateNonce()
-	if err != nil {
-		return err
-	}
-
-	err = p.validateInstID()
-	if err != nil {
-		return err
+func partitionIDToString(pid int32) string {
+	if pid <= 0 {
+		return "nspe"
 	}
-
-	// There seem to be no reasons for an explicit validateVSI
-
-	return nil
+	return "spe"
 }
 
-func (p *Claims) validateProfile() error {
-	v := p.Profile
-
-	if v != nil {
-		profile, err := v.Get()
-		if err != nil {
-			return fmt.Errorf("error extracting profile: %w", err)
-		}
+// Claims is the wrapper around PSA claims
+// nolint: golint
+type Claims struct {
+	Profile           *eat.Profile  `cbor:"18,keyasint" json:"profile"`
+	PartitionID       *int32        `cbor:"-75001,keyasint" json:"partition-id"`
+	SecurityLifeCycle *uint16       `cbor:"-75002,keyasint" json:"security-life-cycle"`
+	ImplID            *[]byte       `cbor:"-75003,keyasint" json:"implementation-id"`
+	BootSeed          *[]byte       `cbor:"-75004,keyasint" json:"boot-seed"`
+	HwVersion         *string       `cbor:"-75005,keyasint,omitempty" json:"hardware-version,omitempty"`
+	SwComponents      []SwComponent `cbor:"-75006,keyasint,omitempty" json:"software-components,omitempty"`
+	NoSwMeasurements  uint          `cbor:"-75007,keyasint,omitempty" json:"no-software-measurements,omitempty"`
+	Nonce             *eat.Nonce    `cbor:"10,keyasint" json:"nonce"`
```

Comment:
Going by the logic of incremental CBOR Tag numbering, would it not be better to put Nonce and InstID tags first and then bring Profile in the sequence?

Reply:
ok